src/desktop/apps/auction/components/DOM.js

```diff
@@ -46,9 +48,29 @@ class DOM extends Component {
     this.$registerBtn.off("click", this.handleRegister)
   }
 
+  showModal = type => {
+    this.props.dispatch(showModal(type))
+  }
+
+  maybeShowModal() {
+    const { pathname } = location
+    console.log(pathname)
+    if (pathname.match(/registration-flow$/) && this.props.me) {
+      this.showModal("RegistrationFlow")
+    } else if (pathname.match(/\/confirm-registration/) && this.props.me) {
+      this.showModal(
+        location.search.match("origin=bid")
+          ? "ConfirmBidAndRegistration"
+          : "ConfirmRegistration"
+      )
+    }
```

Comment:
Getting all the magic modal strings consolidated to one function on the client.

Reply:
Another way to write this would be to use a switch

```jsx
switch (true) {
  case pathname.match(/registration-flow$/) && this.props.me: {
    ...
  }
  case pathname.match(/\/confirm-registration/): {
    ...
  } 
}
```
(Just food for thought)